--- i/include/tbb/internal/_warning_suppress_disable_notice.h
+++ w/include/tbb/internal/_warning_suppress_disable_notice.h
@@ -14,14 +14,8 @@
     limitations under the License.
 */
 
-#if __TBB_SUPPRESS_INTERNAL_DEPRECATED_MESSAGES
-
-#if __INTEL_COMPILER || _MSC_VER
-#pragma warning( pop )
-#elif __GNUC__
-#pragma GCC diagnostic pop
-#elif __clang__
+#if __clang__
 #pragma clang diagnostic pop
+#elif __INTEL_COMPILER || _MSC_VER
+#pragma warning( pop )
 #endif
-
-#endif  // __TBB_SUPPRESS_INTERNAL_DEPRECATED_MESSAGES
--- i/include/tbb/internal/_warning_suppress_enable_notice.h
+++ w/include/tbb/internal/_warning_suppress_enable_notice.h
@@ -16,17 +16,11 @@
 
 #include "../tbb_config.h"
 
-#if __TBB_SUPPRESS_INTERNAL_DEPRECATED_MESSAGES
-
-#if _MSC_VER || __INTEL_COMPILER
-#pragma warning( push )
-#pragma warning( disable: 4996 )
-#elif __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#elif __clang__
+#if __clang__
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#pragma clang diagnostic ignored "-Wdeprecated-volatile"
+#elif _MSC_VER || __INTEL_COMPILER
+#pragma warning( push )
+#pragma warning( disable: 4996 )
 #endif
-
-#endif  // __TBB_SUPPRESS_INTERNAL_DEPRECATED_MESSAGES
