--- i/src/tbb/intrusive_list.h
+++ w/src/tbb/intrusive_list.h
@@ -76,12 +76,12 @@ class intrusive_list_base {
             return my_pos = &node(val);
         }
 
-        bool operator == ( const Iterator& it ) const {
-            return my_pos == it.my_pos;
+        friend bool operator == ( const Iterator& lhs, const Iterator& rhs ) noexcept {
+            return lhs.my_pos == rhs.my_pos;
         }
 
-        bool operator != ( const Iterator& it ) const {
-            return my_pos != it.my_pos;
+        friend bool operator != ( const Iterator& lhs, const Iterator& rhs ) noexcept {
+            return lhs.my_pos != rhs.my_pos;
         }
 
         Iterator& operator++ () {
